releasenotes/notes/reduce-binary-polynomial-71ae7d9a79fb0cf1.yaml

```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Added `dimiod.higherorder.utils.reduce_binary_polynomial` that factors out a common basis for `dimiod.higherorder.utils.make_quadratic` and a future `dimiod.higherorder.utils.make_quadratic_cqm`. Given a `BinaryPolynomial` it return a list of reduced terms with auxillary variables and a list of constraints.
```

Comment:
`dimiod` <- typo. While we're at it, should use ``` `` ``` around code and can just say ``` ``reduce_binary_polynomial()`` ``` rather than the full path IMO since it gets imported into the top level namespace.

Reply:
Done